pkg/engine/events.go

```diff
@@ -605,3 +487,99 @@ func (e *eventEmitter) diagErrorEvent(d *diag.Diag, prefix, msg string, ephemera
 func (e *eventEmitter) diagWarningEvent(d *diag.Diag, prefix, msg string, ephemeral bool) {
 	diagEvent(e, d, prefix, msg, diag.Warning, ephemeral)
 }
+
+func filterResourceProperties(m resource.PropertyMap, debug bool) resource.PropertyMap {
```

Comment:
Happy to make a more targeted fix than this refactoring if reviewers feel that constraining these changes is warranted. I felt that the refactor clarified the code given that it removes some reflection and allows us to be more precise in what we expect here.